go.mod

```diff
@@ -8,6 +8,11 @@ require (
 	github.com/golang-commonmark/puny v0.0.0-20180910110745-050be392d8b8 // indirect
 	github.com/golang/protobuf v1.2.0
 	github.com/google/go-cmp v0.2.0
+	github.com/opennota/wd v0.0.0-20180911144301-b446539ab1e7 // indirect
+	github.com/russross/blackfriday v2.0.0+incompatible // indirect
+	github.com/shurcooL/sanitized_anchor_name v0.0.0-20170918181015-86672fcb3f95 // indirect
+	golang.org/x/sync v0.0.0-20181108010431-42b317875d0f // indirect
+	golang.org/x/tools v0.0.0-20181201035826-d0ca3933b724
```

Comment:
`gencli` runs `tools/import` on all Go files added to 1. format the code 2. tidy the imports & remove extraneous ones (to make the generation code simpler)

The imports for diff lines 12-14 I believe were added because of other code in the project that uses them. I ran `go mod tidy` to add my deps (`tools`) and the others were added too.

I can remove the others if they aren't necessary

Reply:
Hmm. I see what you mean. How hard would it be to only add imports we actually need?

This came up when I worked on `gapic-generator-go` earlier. The problem was that eventually you create a bug where the generator prints syntactically invalid code. Then `tools/import` can't work on the code and things become frustrating to debug.

Reply:
There aren't a lot of imports, so I can take another look at this.

Personally, I like formatting the generated code prior to emission. I'd prefer to not generate anything in the case where something comes out broken rather than wait for another step to catch it (i.e. attempting `go build` on the generated code).

I do sympathize with the difficulty in debugging such errors. No code gets generated to debug! 

Wdyt about an option to disable formatting so that a failed run could be debugged with a subsequent `--gcli_opt 'fmt:false` run?

I'm not married to formatting, so I don't mind throwing it out altogether, but would like to see it stay in some capacity if possible because I find it useful.

Reply:
Another part of my thinking is that if an import is used sometimes, it sounds cleaner for the program to know when to import it than to generate too many imports and strip out later. One more thing to watch out for is that the result of `go/format` which the `import` library uses can change with version, so you probably want to do baseline test before formatting.

I understand your argument though; I can be convince either way. If we end up doing the wrong thing, it isn't that hard to switch. I think the flag you suggested sounds reasonable.